Testsuite crashes due to unaligned access on sparc64

The testsuite crashes on sparc64 due to unaligned access:

```
make  check-TESTS
make[4]: Entering directory '/<<PKGBUILDDIR>>/test'
make[5]: Entering directory '/<<PKGBUILDDIR>>/test'
FAIL: run_tests03.pl
PASS: run_tests06.pl
FAIL: run_tests00.pl
FAIL: run_tests02.pl
FAIL: run_tests05.pl
FAIL: run_tests01.pl
FAIL: run_tests04.pl
PASS: run_tests09.pl
PASS: run_tests08.pl
PASS: run_tests12.pl
PASS: run_tests07.pl
PASS: run_tests10.pl
FAIL: run_tests11.pl
PASS: run_tests13.pl
PASS: run_tests15.pl
PASS: run_tests14.pl
========================================
   pmix 3.1.4rc2: test/test-suite.log
========================================

# TOTAL: 16
# PASS:  9
# SKIP:  0
# XFAIL: 0
# FAIL:  7
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: run_tests00.pl
====================

./pmix_test -n 4 --ns-dist 3:1 --fence "[db | 0:0-2;1:0]" 2>&1
Bus error

CODE 35328
FAIL run_tests00.pl (exit status: 138)

FAIL: run_tests01.pl
====================

./pmix_test -n 4 --ns-dist 3:1 --fence "[db | 0:;1:0]" 2>&1
Bus error

CODE 35328
FAIL run_tests01.pl (exit status: 138)

FAIL: run_tests02.pl
====================

./pmix_test -n 4 --ns-dist 3:1 --fence "[db | 0:;1:]" 2>&1
Bus error

CODE 35328
FAIL run_tests02.pl (exit status: 138)

FAIL: run_tests03.pl
====================

./pmix_test -n 4 --ns-dist 3:1 --fence "[0:]" 2>&1
Bus error

CODE 35328
FAIL run_tests03.pl (exit status: 138)

FAIL: run_tests04.pl
====================

./pmix_test -n 4 --ns-dist 3:1 --fence "[b | 0:]" 2>&1
Bus error

CODE 35328
FAIL run_tests04.pl (exit status: 138)

FAIL: run_tests05.pl
====================

./pmix_test -n 4 --ns-dist 3:1 --fence "[d | 0:]" --noise "[0:0,1]" 2>&1
Bus error

CODE 35328
FAIL run_tests05.pl (exit status: 138)

FAIL: run_tests11.pl
====================

./pmix_test -n 5 --test-resolve-peers --ns-dist "1:2:2" 2>&1
Bus error

CODE 35328
FAIL run_tests11.pl (exit status: 138)

============================================================================
Testsuite summary for pmix 3.1.4rc2
============================================================================
# TOTAL: 16
# PASS:  9
# SKIP:  0
# XFAIL: 0
# FAIL:  7
# XPASS: 0
# ERROR: 0
============================================================================
See test/test-suite.log
Please report to http://pmix.github.io/master
============================================================================
```

Full build log available at: https://buildd.debian.org/status/fetch.php?pkg=pmix&arch=sparc64&ver=3.1.4-1&stamp=1577756718&raw=0

Both the master and the v3.1 branch still crash due to unaligned access:

```
FAIL: run_tests00.pl
FAIL: run_tests01.pl
FAIL: run_tests02.pl
FAIL: run_tests03.pl
FAIL: run_tests04.pl
FAIL: run_tests05.pl
PASS: run_tests06.pl
PASS: run_tests07.pl
PASS: run_tests08.pl
PASS: run_tests09.pl
PASS: run_tests10.pl
FAIL: run_tests11.pl
PASS: run_tests12.pl
PASS: run_tests13.pl
PASS: run_tests14.pl
PASS: run_tests15.pl
============================================================================
Testsuite summary for pmix 3.1.5rc1
============================================================================
# TOTAL: 16
# PASS:  9
# SKIP:  0
# XFAIL: 0
# FAIL:  7
# XPASS: 0
# ERROR: 0
============================================================================
See test/test-suite.log
Please report to http://pmix.github.io/master
============================================================================
make[4]: *** [Makefile:1237: test-suite.log] Error 1
make[4]: Leaving directory '/home/glaubitz/openpmix/test'
make[3]: *** [Makefile:1345: check-TESTS] Error 2
make[3]: Leaving directory '/home/glaubitz/openpmix/test'
make[2]: *** [Makefile:1549: check-am] Error 2
make[2]: Leaving directory '/home/glaubitz/openpmix/test'
make[1]: *** [Makefile:1129: check-recursive] Error 1
make[1]: Leaving directory '/home/glaubitz/openpmix/test'
make: *** [Makefile:745: check-recursive] Error 1
glaubitz@gcc202:~/openpmix$ ./test/run_tests00.pl
/usr/bin/timeout --preserve-status -k 35 30  ./pmix_test -n 4 --ns-dist 3:1 --fence "[db | 0:0-2;1:0]" 2>&1
Bus error

CODE 35328
glaubitz@gcc202:~/openpmix$
```

Can you tell me how to run the ```pmix_test``` binary in GDB, so I can figure out where the unaligned access happens? Most likely it's just a matter of invalid pointer casting, i.e. something like explained here: https://wiki.sei.cmu.edu/confluence/display/c/EXP36-C.+Do+not+cast+pointers+into+more+strictly+aligned+pointer+types

--- a/test/test_server.c
+++ b/test/test_server.c
@@ -247,7 +247,7 @@ static void server_unpack_procs(char *buf, size_t size)
     char *nspace;
 
     while ((size_t)(ptr - buf) < size) {
-        ns_count = *(size_t *)ptr;
+        memcpy (&ns_count, ptr, sizeof(size_t));
         ptr += sizeof(size_t);
 
         for (i = 0; i < ns_count; i++) {
@@ -255,16 +255,16 @@ static void server_unpack_procs(char *buf, size_t size)
             size_t ltasks, ntasks;
             int server_id;
 
-            server_id = *(int *)ptr;
+            memcpy (&server_id, ptr, sizeof(int));
             ptr += sizeof(int);
 
             nspace = ptr;
             ptr += PMIX_MAX_NSLEN+1;
 
-            ntasks = *(size_t *)ptr;
+            memcpy (&ntasks, ptr, sizeof(size_t));
             ptr += sizeof(size_t);
 
-            ltasks = *(size_t *)ptr;
+            memcpy (&ltasks, ptr, sizeof(size_t));
             ptr += sizeof(size_t);
 
             PMIX_LIST_FOREACH(tmp, server_nspace, server_nspace_t) {
@@ -286,7 +286,8 @@ static void server_unpack_procs(char *buf, size_t size)
             }
             size_t i;
             for (i = 0; i < ltasks; i++) {
-                int rank = *(int *)ptr;
+                int rank;
+                memcpy (&rank, ptr, sizeof(int));
                 ptr += sizeof(int);
                 if (ns_item->task_map[rank] >= 0) {
                     continue;

